### ISSUE TITLE
conv corner case 

### ISSUE THREAD

**Issue by KristofferC** [JuliaDSP/DSP.jl#168]
Moved from https://github.com/JuliaLang/julia/issues/20539

`conv( randn(5), randn(0) )`, say
should be, i'd think, should be a vector of length 0
right now it's an error

`conv(randn(0), randn(0))` is trickier, but a vector length 0 is not unreasonable

**Comment by RuiRojo** [comment]
Does it have any use case? Otherwise, I think an error seems better, right?
Without that definition, you know `length(conv(u, v)) === length(u) + length(v) = 1`.

**Comment by martinholters** [comment]
Yep, `length(conv(u, v)) == length(u) + length(v) - 1` is a nice property which we can, do, and I'd say should, preserve even if one of `u` and `v` is empty. For `length(u)==length(v)==0`, the best we can do is probably returning an empty vector, which indeed we do.

However, `conv(randn(5), randn(0); algorithm=:fft)` still throws, which it shouldn't.

**Comment by martinholters** [comment]
Don't know, haven't looked into it. The error originates in `optimalfftfiltlength`, so fixing that might be the minimal bugfix, provided the FFT based filtering then works.

**Comment by martinholters** [comment]
Right, if I'm not mistaken, `:direct` will reduce to zeroing the output and and an empty loop, so that's good. OTOH, I've just tried patching `optimalfftfiltlength` so that it doesn't throw, but then `unsafe_conv_kern_os!` or `_conv_kern_fft!` throw (depending on which `:fft_` variant is chosen), so that route is not as simple.
